platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/EventIntake.java

```diff
@@ -163,66 +163,70 @@ public void addUnlinkedEvent(final GossipEvent event) {
+                            event::toMediumString);
+                }
             }
-        }
 
-        if (prehandlePool == null) {
-            // Prehandle transactions on the intake thread (i.e. this thread).
-            prehandleEvent.accept(event);
-        } else {
-            // Prehandle transactions on the thread pool.
-            prehandlePool.submit(buildPrehandleTask(event));
-        }
+            if (prehandlePool == null) {
+                // Prehandle transactions on the intake thread (i.e. this thread).
+                prehandleEvent.accept(event);
+            } else {
+                // Prehandle transactions on the thread pool.
+                prehandlePool.submit(buildPrehandleTask(event));
+            }
 
-        // record the event in the hashgraph, which results in the events in consEvent reaching consensus
-        final List<ConsensusRound> consRounds = consensusWrapper.addEvent(event, addressBook);
-        // #5762 after we calculate roundCreated, se set its value in GossipEvent so that it can be shared with other
-        // nodes
-        event.getBaseEvent().setRoundCreated(event.getRoundCreated());
-        stats.addedToConsensus();
-        dispatcher.eventAdded(event);
-        stats.dispatchedAdded();
-        if (consRounds != null) {
-            consRounds.forEach(this::handleConsensus);
-            stats.dispatchedRound();
-        }
-        if (consensus().getMinGenerationNonAncient() > minGenNonAncientBeforeAdding) {
-            // consensus rounds can be null and the minNonAncient might change, this is probably because of a round
-            // with no consensus events, so we check the diff in generations to look for stale events
-            handleStale(minGenNonAncientBeforeAdding);
-            stats.dispatchedStale();
+            // record the event in the hashgraph, which results in the events in consEvent reaching consensus
+            final List<ConsensusRound> consRounds = consensusWrapper.addEvent(event, addressBook);
+            // #5762 after we calculate roundCreated, we set its value in GossipEvent so that it can be shared with
+            // other nodes
+            event.getBaseEvent().setRoundCreated(event.getRoundCreated());
+            stats.addedToConsensus();
+            dispatcher.eventAdded(event);
+            stats.dispatchedAdded();
+            if (consRounds != null) {
+                consRounds.forEach(this::handleConsensus);
+                stats.dispatchedRound();
+            }
+            if (consensus().getMinGenerationNonAncient() > minGenNonAncientBeforeAdding) {
+                // consensus rounds can be null and the minNonAncient might change, this is probably because of a round
+                // with no consensus events, so we check the diff in generations to look for stale events
+                handleStale(minGenNonAncientBeforeAdding);
+                stats.dispatchedStale();
+            }
+            stats.doneIntakeAddEvent();
```

Comment:
does the stats update need to be part of the finally block?  An intermediate exception would leave the `stats.startIntakeAddEvent()` unpaired.

Reply:
I think you're probably right - however there is an issue I will be working this sprint or next to rework intake stats https://github.com/hashgraph/hedera-services/issues/8129

I tend to favor leaving the stats stuff as-is for this PR, rather than making changes to a system that's on its way out the door